default-plugins/strider/src/main.rs

```diff
@@ -24,37 +30,66 @@ impl ZellijPlugin for State {
                     *self.selected_mut() = min(self.files.len() - 1, next);
                 }
                 Key::Right | Key::Char('\n') | Key::Char('l') if !self.files.is_empty() => {
-                    match self.files[self.selected()].clone() {
-                        FsEntry::Dir(p, _) => {
-                            self.path = p;
-                            refresh_directory(self);
-                        }
-                        FsEntry::File(p, _) => open_file(&p),
-                    }
+                    self.traverse_dir_or_open_file();
+                    self.ev_history.clear();
                 }
                 Key::Left | Key::Char('h') => {
                     self.path.pop();
                     refresh_directory(self);
                 }
-
                 Key::Char('.') => {
                     self.toggle_hidden_files();
                     refresh_directory(self);
                 }
 
                 _ => (),
-            };
+            },
+            Event::Mouse(mouse_event) => match mouse_event {
+                Mouse::ScrollDown(_) => {
+                    let next = self.selected().saturating_add(1);
+                    *self.selected_mut() = min(self.files.len() - 1, next);
+                }
+                Mouse::ScrollUp(_) => {
+                    *self.selected_mut() = self.selected().saturating_sub(1);
+                }
+                Mouse::MouseRelease(Some((line, _))) => {
+                    if line < 0 {
+                        return;
+                    }
+                    let mut should_select = true;
+                    if let Some((Event::Mouse(Mouse::MouseRelease(Some((prev_line, _)))), t)) =
+                        prev_event
+                    {
+                        if prev_line == line
+                            && Instant::now().saturating_duration_since(t).as_millis() < 400
+                        {
+                            self.traverse_dir_or_open_file();
+                            self.ev_history.clear();
+                            should_select = false;
+                        }
+                    }
+                    if should_select && self.scroll() + (line as usize) < self.files.len() {
```

Comment:
I've noticed that `line` is an `isize`. Do you know the reason for that? If we we ever get a `line < 0`, these `as usize` calls could cause quite a bit of chaos (I've been bit by that myself recently).